cpp/daal/src/externals/service_stat_ref.h

```diff
@@ -0,0 +1,328 @@
+#define __DAAL_VSL_SS_2R_MOM                        0x0000000000000002
+#define __DAAL_VSL_SS_2C_MOM                        0x0000000000000010
+#define __DAAL_VSL_SS_2C_SUM                        0x0000000020000000
+#define __DAAL_VSL_SS_VARIATION                     0x0000000000000200
+#define __DAAL_VSL_SS_ED_ACCUM_WEIGHT               23
+#define __DAAL_VSL_SS_METHOD_FAST_USER_MEAN         0x00000100
+#define __DAAL_VSL_SS_METHOD_BACON_MEDIAN_INIT      0x00000002
+#define __DAAL_VSL_SS_METHOD_BACON_MAHALANOBIS_INIT 0x00000001
+#define __DAAL_VSL_SS_OUTLIERS                      0x0000000000080000
+#define __DAAL_VSL_SS_METHOD_BACON                  0x00000020
+#define __DAAL_VSL_SS_QUANTS                        0x0000000000010000
+#define __DAAL_VSL_SS_ED_QUANT_ORDER_N              24
+#define __DAAL_VSL_SS_ED_QUANT_ORDER                25
+#define __DAAL_VSL_SS_ED_QUANT_QUANTILES            26
+#define __DAAL_VSL_SS_SORTED_OBSERV                 0x0000008000000000
+#define __DAAL_VSL_SS_ED_SORTED_OBSERV              78
+#define __DAAL_VSL_SS_ED_SORTED_OBSERV_STORAGE      79
+#define __DAAL_VSL_SS_METHOD_RADIX                  0x00100000
+
+#define __DAAL_VSL_SS_ERROR_BAD_QUANT_ORDER       -4022
+#define __DAAL_VSL_SS_ERROR_INDICES_NOT_SUPPORTED -4085
+
+    typedef void (*threadfuncfor)(DAAL_INT, DAAL_INT, void *, func_type);
+    typedef void (*threadfuncforordered)(DAAL_INT, DAAL_INT, void *, func_type);
+    typedef void (*threadfuncsection)(DAAL_INT, void *, func_type);
+    typedef void (*threadfuncordered)(DAAL_INT, DAAL_INT, DAAL_INT, void *, func_type);
+    typedef DAAL_INT (*threadgetlimit)(void);
+
+    struct ThreadingFuncs
+    {
+        threadfuncfor funcfor;
+        threadfuncfor funcforordered;
+        threadfuncsection funcsection;
+        threadfuncordered funcordered;
+        threadgetlimit getlimit;
+    };
+
+    static void _daal_mkl_threader_for_sequential(DAAL_INT n, DAAL_INT threads_request, void * a, func_type func)
+    {
+        DAAL_INT i;
+
+        for (i = 0; i < n; i++)
+        {
+            func(i, 0, 1, a);
+        }
+    }
+
+    static void _daal_mkl_threader_for_ordered_sequential(DAAL_INT n, DAAL_INT threads_request, void * a, func_type func)
+    {
+        DAAL_INT i;
+
+        for (i = 0; i < n; i++)
+        {
+            func(i, 0, 1, a);
+        }
+    }
+
+    static void _daal_mkl_threader_sections_sequential(DAAL_INT threads_request, void * a, func_type func)
+    {
+        func(0, 0, 1, a);
+    }
+
+    static void _daal_mkl_threader_ordered_sequential(DAAL_INT i, DAAL_INT th_idx, DAAL_INT th_num, void * a, func_type func)
+    {
+        func(i, th_idx, th_num, a);
+    }
+
+    static DAAL_INT _daal_mkl_threader_get_max_threads_sequential()
+    {
+        return 1;
+    }
+
+    static void _daal_mkl_threader_for(DAAL_INT n, DAAL_INT threads_request, void * a, func_type func)
+    {
+        //fpk_vsl_serv_threader_for(n, threads_request, a, func);
```

Comment:
According to library rules, as I remember, raising an exception is prohibited here. It would be good to separate threading changes from backend changes.